lib/utils/isStandardSyntaxTypeSelector.js

```diff
@@ -36,6 +36,23 @@ module.exports = function(node /*: Object*/) /*: boolean*/ {
     return false;
   }
 
+  let normalizedValue = node.value;
+
+  // Ignore operators before variables (example -$variable)
+  if (/^[-+*/]/.test(node.value[0])) {
+    normalizedValue = normalizedValue.slice(1);
+  }
+
+  // SCSS variable
```

Comment:
I think we need to look into this issue more closely.

I think we need to:

- establish how we test against a non-standard custom syntax i.e. `postcss-jsx` in this instance
- confirm exactly what syntax is causing the original bug

The first one could involve a change to `testRule` so that the `syntax` property accepts a parser function as well as a string. We'd then include `postcss-jsx` as a dev dependency.

The latter one needs an investigation into what non-standard syntax can be used in a selector. For example, I'm pretty sure that within Scss and Less you can only use variables within interpolation, which makes those conditionals in the `isStandard*` util redundant.

Additionally, it might be worth waiting until https://github.com/stylelint/stylelint/pull/3284/files is merged before revisiting this issue as version 4 of the selector parser is less likley to categorize these things as `tag` selectors.

Reply:
In which cases is the percent sign used before the selector?

Reply:
@jeddy3 I agree with your opinion. Shall we put it off for the present?